File: scripts/generate-purple-favicon.py
#!/usr/bin/env python3
import os
from PIL import Image, ImageDraw, ImageFont
import io

def create_purple_favicon(size):
    """Create a purple QC favicon at the specified size."""
    # Create a new image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Define purple gradient colors
    purple_dark = (107, 70, 193)  # #6b46c1
    purple_mid = (124, 58, 237)   # #7c3aed
    purple_light = (139, 92, 246)  # #8b5cf6
    
    # Draw rounded rectangle background
    radius = int(size * 0.15)
    
    # Create a rounded rectangle
    draw.rounded_rectangle(
        [(0, 0), (size-1, size-1)],
        radius=radius,
        fill=purple_mid
    )
    
    # Draw text
    text = "QC"
    
    # Try to use a good font, fallback to default if not available
    font_size = int(size * 0.4)
    if size <= 32:
        font_size = int(size * 0.45)
    
    try:
        # Try to use a system font
        font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
    except:
        # Fallback to default font
        font = ImageFont.load_default()
    
    # Get text bbox for centering
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    # Center the text
    x = (size - text_width) // 2
    y = (size - text_height) // 2 - bbox[1]
    
    # Draw white text
    draw.text((x, y), text, fill=(255, 255, 255), font=font)
    
    return img

def generate_favicons():
    """Generate all required favicon sizes."""
    sizes = {
        'favicon-16x16.png': 16,
        'favicon-32x32.png': 32,
        'favicon-64x64.png': 64,
        'logo192.png': 192,
        'logo512.png': 512
    }
    
    output_dir = os.path.join(os.path.dirname(__file__), '..', 'public')
    
    for filename, size in sizes.items():
        img = create_purple_favicon(size)
        filepath = os.path.join(output_dir, filename)
        img.save(filepath, 'PNG')
        print(f"Created {filename} ({size}x{size})")
    
    # Create ICO file with multiple sizes
    ico_sizes = [(16, 16), (32, 32), (48, 48), (64, 64)]
    ico_images = []
    for size in ico_sizes:
        ico_images.append(create_purple_favicon(size[0]))
    
    ico_path = os.path.join(output_dir, 'favicon.ico')
    ico_images[0].save(ico_path, format='ICO', sizes=ico_sizes)
    print(f"Created favicon.ico with sizes: {ico_sizes}")

if __name__ == "__main__":
    try:
        generate_favicons()
        print("\nAll favicons generated successfully!")
    except Exception as e:
        print(f"Error generating favicons: {e}")
        print("\nTrying alternative method...")
        
        # Alternative method using basic shapes if PIL features are limited
        import numpy as np
        from PIL import Image
        
        def create_simple_favicon(size):
            # Create purple background
            img = Image.new('RGB', (size, size), (124, 58, 237))
            draw = ImageDraw.Draw(img)
            
            # Draw white "QC" text
            text = "QC"
            font_size = int(size * 0.35)
            
            # Use basic font
            font = ImageFont.load_default()
            
            # Estimate text position (centered)
            x = size // 4
            y = size // 3
            
            draw.text((x, y), text, fill=(255, 255, 255), font=font)
            
            return img
        
        # Generate basic versions
        output_dir = os.path.join(os.path.dirname(__file__), '..', 'public')
        
        # At least create the main favicon.ico
        img32 = create_simple_favicon(32)
        img32.save(os.path.join(output_dir, 'favicon.ico'), 'ICO')
        print("Created basic favicon.ico")
        
        # Create basic PNGs
        for name, size in [('logo192.png', 192), ('logo512.png', 512)]:
            img = create_simple_favicon(size)
            img.save(os.path.join(output_dir, name), 'PNG')
            print(f"Created basic {name}")

File: scripts/generate-purple-favicon.js
const fs = require('fs');
const path = require('path');

// Simple SVG favicon generator
function createPurpleFaviconSVG(size) {
    const radius = size * 0.15;
    const fontSize = size <= 32 ? size * 0.45 : size * 0.4;
    
    return `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${size}" height="${size}" viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="purpleGradient" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#6b46c1;stop-opacity:1" />
      <stop offset="50%" style="stop-color:#7c3aed;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#8b5cf6;stop-opacity:1" />
    </linearGradient>
  </defs>
  
  <!-- Background with rounded corners -->
  <rect width="${size}" height="${size}" rx="${radius}" fill="url(#purpleGradient)"/>
  
  <!-- Subtle highlight on top -->
  <rect width="${size}" height="${size * 0.5}" rx="${radius}" fill="white" opacity="0.1"/>
  
  <!-- QC Text -->
  <text x="${size/2}" y="${size/2 + fontSize * 0.1}" 
        font-family="-apple-system, BlinkMacSystemFont, 'SF Pro Display', 'Helvetica Neue', Arial, sans-serif" 
        font-size="${fontSize}" 
        font-weight="bold"
        text-anchor="middle" 
        fill="white">QC</text>
</svg>`;
}

// Create a simple purple SVG icon that we'll use for now
const purpleIconSVG = `<?xml version="1.0" encoding="UTF-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="purpleGradient" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#6b46c1;stop-opacity:1" />
      <stop offset="50%" style="stop-color:#7c3aed;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#8b5cf6;stop-opacity:1" />
    </linearGradient>
  </defs>
  
  <!-- Background with rounded corners -->
  <rect width="512" height="512" rx="80" fill="url(#purpleGradient)"/>
  
  <!-- Subtle highlight on top -->
  <rect width="512" height="256" rx="80" fill="white" opacity="0.1"/>
  
  <!-- QC Text -->
  <text x="256" y="280" 
        font-family="-apple-system, BlinkMacSystemFont, 'SF Pro Display', 'Helvetica Neue', Arial, sans-serif" 
        font-size="200" 
        font-weight="bold"
        text-anchor="middle" 
        fill="white">QC</text>
</svg>`;

// Save the SVG files
const publicDir = path.join(__dirname, '..', 'public');

// Create different sizes of SVG files
const sizes = [16, 32, 192, 512];
sizes.forEach(size => {
    const filename = size < 100 ? `favicon-${size}x${size}.svg` : `logo${size}.svg`;
    const svgContent = createPurpleFaviconSVG(size);
    fs.writeFileSync(path.join(publicDir, filename), svgContent);
    console.log(`Created ${filename}`);
});

// Also save the main purple icon
fs.writeFileSync(path.join(publicDir, 'purple-icon.svg'), purpleIconSVG);
console.log('Created purple-icon.svg');

console.log('\nSVG favicons created successfully!');
console.log('Note: You may need to convert these to PNG/ICO format using an online converter or image editing software.');